huaweicloud/services/apig/resource_huaweicloud_apig_instance.go

```diff
@@ -308,17 +321,23 @@ func resourceInstanceCreate(ctx context.Context, d *schema.ResourceData, meta in
 	}
 	d.SetId(resp.Id)
 
+	instanceId := d.Id()
 	stateConf := &resource.StateChangeConf{
 		Pending:      []string{"PENDING"},
 		Target:       []string{"COMPLETED"},
-		Refresh:      InstanceStateRefreshFunc(client, d.Id(), []string{"Running"}),
+		Refresh:      InstanceStateRefreshFunc(client, instanceId, []string{"Running"}),
 		Timeout:      d.Timeout(schema.TimeoutCreate),
 		Delay:        20 * time.Second,
 		PollInterval: 20 * time.Second,
 	}
 	_, err = stateConf.WaitForStateContext(ctx)
 	if err != nil {
-		return diag.Errorf("error waiting for the dedicated instance (%s) to become running: %s", d.Id(), err)
+		return diag.Errorf("error waiting for the dedicated instance (%s) to become running: %s", instanceId, err)
+	}
+
+	err = instances.UpdateTags(client, buildTagsUpdateOpts(d.Get("tags").(map[string]interface{}), instanceId, "create"))
+	if err != nil {
+		log.Printf("[WARN] error creating instance tags: %s", err)
```

Comment:
this error should be returned

Reply:
For the create method, the error return is not good because some region will not support this function (API).

Reply:
emmm，maybe add a `d.GetOk()` is better.